src/coreclr/jit/gentree.h

```diff
@@ -543,6 +543,7 @@ enum GenTreeFlags : unsigned int
     GTF_QMARK_CAST_INSTOF       = 0x80000000, // GT_QMARK -- Is this a top (not nested) level qmark created for
                                               //             castclass or instanceof?
 
+    GTF_BOX_CLONED              = 0x40000000, // GT_BOX -- this box and its operand has been cloned, cannot assume it to be single-use anymore
```

Comment:
The local under the box is also marked as "cloned" -- is a new flag truly needed?

Reply:
Transformations can affect that local in arbitrary ways (I mentioned it above).

Reply:
Hmm, yes, sorry I missed the description somehow.

I suppose this is more robust in the fact of something like:
```cs
var sdsuTmp = boxTmp; // Cloned.
Use(BOX(sdsuTmp));
```
So it is needed.

Reply:
Right, something like that. It would probably be ok in practice, especially since `gtTryRemoveBoxUpstreamEffects` is only called very early, but given the function name it seems best to make it more robust.